UnityProject/Assets/Scripts/UI/Systems/Lobby/CharacterCustomization.cs

```diff
@@ -1511,6 +1475,120 @@ private void RefreshRace()
+		{
+			CharacterSettings random = new CharacterSettings();
+			// Randomise gender
+			Type gender = typeof(BodyType);
+			Array genders = gender.GetEnumValues();
+			int index = UnityEngine.Random.Range(0,3);
+			random.BodyType = (BodyType)genders.GetValue(index);
+
+			//Randomises player name and age and skin tone.
+			random.Name = RandomizeCharacterName(random);
+			random.Age  = UnityEngine.Random.Range(19, 78);
+			random.SkinTone = RandomizeCharacterSkinToneHTMLString(random);
+			random.SerialisedExternalCustom = GetRandomUnderwearCustomisation(random);
+
+			//Randomises player accents. (Italian, Scottish, etc)
+			random.Speech = RandomizeCharachterAccent();
+
+			return random;
+		}
+
+		public static List<ExternalCustomisation> GetRandomUnderwearCustomisation(CharacterSettings data)
+		{
+			var RaceData = GetRaceData(data);
+			List<ExternalCustomisation> externalCustomisations = new List<ExternalCustomisation>();
+
+			void logic(CustomisationAllowedSetting setting)
+			{
+				PlayerCustomisationData customizationToAdd = setting.CustomisationGroup.PlayerCustomisations.PickRandom();
+				ExternalCustomisation newExternalCustomisation = new ExternalCustomisation();
+				newExternalCustomisation.Key = customizationToAdd.name;
+				newExternalCustomisation.SerialisedValue = SerialiseCustomizationData(customizationToAdd);
+				externalCustomisations.Add(newExternalCustomisation);
+			}
+
+			foreach (CustomisationAllowedSetting customisation in RaceData.Base.CustomisationSettings)
+			{
+				if (customisation.CustomisationGroup.name == "PlayerUnderShirt") logic(customisation);
+				if (customisation.CustomisationGroup.name == "PlayerUnderWear") logic(customisation);
+				if (customisation.CustomisationGroup.name == "PlayerSocks") logic(customisation);
+			}
+
+			return externalCustomisations;
+		}
+
+		public static CharacterSettings.CustomisationClass SerialiseCustomizationData(PlayerCustomisationData data)
+		{
+			var newcurrentSetting = new CharacterSettings.CustomisationClass();
+			newcurrentSetting.Colour = "#" + ColorUtility.ToHtmlStringRGB(new Color(UnityEngine.Random.Range(0.1f, 1f),
+				UnityEngine.Random.Range(0.1f, 1f),
+				UnityEngine.Random.Range(0.1f, 1f), 1f));
+			newcurrentSetting.SelectedName = data.Name;
+			return newcurrentSetting;
+
+		}
+
+		public static string RandomizeCharacterSkinToneHTMLString(CharacterSettings data)
+		{
+			var RaceData = GetRaceData(data);
+
+			List<Color> raceSkinColors = RaceData.Base.SkinColours;
+			if (raceSkinColors.Count != 0)
+			{
+				return "#" + ColorUtility.ToHtmlStringRGB(raceSkinColors[UnityEngine.Random.Range(0, raceSkinColors.Count - 1)]);
+			}
+			return "#" + ColorUtility.ToHtmlStringRGBA(new Color(UnityEngine.Random.Range(0.1f, 1f),
+					UnityEngine.Random.Range(0.1f, 1f),
+					UnityEngine.Random.Range(0.1f, 1f), 1f));
+		}
+
+		public static string RandomizeCharacterName(CharacterSettings data, bool isNotPlayerDebug = true)
```

Comment:
I think it might be good idea to move the stuff within character settings,
so you can just do character settings. Randomise, 
You wouldn't be able do this for the body part customisations though

Reply:
Done 👍